python/ray/tune/api_v2/tuner.py

```diff
@@ -0,0 +1,231 @@
+import copy
+import os
+from typing import Any, Callable, Dict, List, Optional, Type, Union
+
+import ray
+import ray.cloudpickle as pickle
+from ray.data import Dataset
+
+from ray.tune import TuneError
+from ray.tune.api_v2.convertible_to_trainable import ConvertibleToTrainable
+from ray.tune.callback import Callback
+from ray.tune.trainable import Trainable
+from ray.tune.tune import run
+from ray.util.client.common import ClientActorHandle
+from ray.util.ml_utils.node import force_on_current_node
+
+TUNER_INTERNAL = "tuner_internal"
+DATASETS = "datasets"
+
+
+class TunerInternal:
+    def __init__(
+        self,
+        restore_path: str = None,
+        trainable: Optional[
+            Union[
+                str,
+                Callable,
+                Type[Trainable],
+                Type[ConvertibleToTrainable],
+                ConvertibleToTrainable,
+            ]
+        ] = None,
+        run_config: Optional[Dict[str, Any]] = None,
+        param_space: Optional[Dict[str, Any]] = None,
+        name: Optional[str] = None,
+        callbacks: Optional[List[Callback]] = None,
+    ):
+        # Start from restore
+        if restore_path:
+            trainable_ckpt = os.path.join(restore_path, "trainable.pkl")
+            with open(trainable_ckpt, "rb") as fp:
+                trainable = pickle.load(fp)
+
+            tuner_ckpt = os.path.join(restore_path, "tuner.pkl")
+            with open(tuner_ckpt, "rb") as fp:
+                tuner = pickle.load(fp)
+                self.__dict__.update(tuner.__dict__)
+
+            self.is_restored = True
+            self.trainable = trainable
+            self.experiment_path = restore_path
+            return
+
+        # Start from fresh
+        assert trainable
```

Comment:
Not needed now but later we should make sure to validate arguments properly and raise actionable error messages

Reply:
Acknowledged.